### ISSUE TITLE
Plugin replace blindly all variables in dockerFile

### ISSUE THREAD

**Issue by michasn** [fabric8io/docker-maven-plugin#894]
### Description
Plugin replace blindly all variables in dockerFile. This issue is related with new functionality provided by https://github.com/fabric8io/docker-maven-plugin/issues/871
### Info
original docker file:
FROM ubuntu
ENV USER some_user
ENV HOME /home/${USER} 

docker file created by the plugin in zipped directory 
FROM ubuntu
ENV USER some_user
ENV HOME /home/jenkins 

Expected behaviour 
docker file created by the plugin in zipped directory 
FROM ubuntu
ENV USER some_user
ENV HOME /home/${USER} 

* d-m-p version :0.23.0

**Comment by scoplin** [comment]
I think you can get around the problem by using a different [build filtering](https://dmp.fabric8.io/#build-filtering) token, e.g. `@`

Other possible changes that could help here:
* Add a standard `\` as an escape syntax.  Should be easy to add and I'm not sure configuration is really needed.
* Make a configurable escape alongside the existing filter prefix/suffix options.  (Dubious necessity?)
* Consider removing environment variable interpolation altogether as it invites problems like this one.

Thoughts?

**Comment by rhuss** [comment]
Yes, a different build filter token to distinguish the different kind of properties should be used. Actually they were introduced exactly for this use case. 

You can use a single char like `@` and then props like `@project.artifactId@` or an expression with a wildcard e.g. `#{*}` which leads to properties like `#{project.artifactId}`

@michasn Does this solve your use case ?

**Comment by stromnet** [comment]
Just a FYI:
Noticed that I have same problem. My dockerfile wants to set (and use) the ENV var JAVA_HOME:

pom.xml
```
<args>
  <VERSION>${java.version}</VERSION>
  <MINOR>${java.minor}</MINOR>
  <UPDATE>${java.update}</UPDATE>
</args>
```
Dockerfile
```
ENV JAVA_HOME=/opt/apps/bin/java/jdk1.${VERSION}.${MINOR}_${UPDATE}
ENV JRE_HOME=${JAVA_HOME}/jre
...
```



Running with verbose:
```
[INFO] DOCKER> ---> a10d316d00ca
[INFO] DOCKER> Step 2/12 : ENV JAVA_HOME=/opt/apps/bin/java/jdk1.${VERSION}.${MINOR}_${UPDATE}
[INFO] DOCKER> 
[INFO] DOCKER> ---> Running in 277d23108185
[INFO] DOCKER> Removing intermediate container 277d23108185
[INFO] DOCKER> ---> 043667d8fb7c
[INFO] DOCKER> Step 3/12 : ENV JRE_HOME=/Library/Java/JavaVirtualMachines/jdk1.8.0_71.jdk/Contents/Home/jre
```

The resulting image is obviously unusable.

Adding `<filter>@@</filter>`, removing `<args>` and then using the following instead, helped:
```
ENV JAVA_HOME=/opt/apps/bin/java/jdk1.@@java.version@@.@@java.minor@@_@@java.update@@
ENV JRE_HOME=${JAVA_HOME}/jre
```

**Comment by NancyChunlan** [comment]
We met the same isue, all of the env properties are used as build args during docker build. 
Although we used @ to filter some values, but we need  env vars inside container to be used, not from outside, from the host.
